sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -471,19 +531,21 @@ case class WholeStageCodegenExec(child: SparkPlan) extends UnaryExecNode with Co
         }
        """, inlineToOuterClass = true)
 
+    val className = generatedClassName()
+
     val source = s"""
       public Object generate(Object[] references) {
-        return new GeneratedIterator(references);
+        return new $className(references);
       }
 
       ${ctx.registerComment(s"""Codegend pipeline for\n${child.treeString.trim}""")}
-      final class GeneratedIterator extends org.apache.spark.sql.execution.BufferedRowIterator {
+      final class $className extends ${classOf[BufferedRowIterator].getName} {
```

Comment:
Can we always add `codegenStageId` as a comment by using ctx.registerComment()?

Reply:
Yes, that's a great idea that I missed! Thanks for your comments, @kiszk !

We can get that effect in two ways:
1. Change the current line `${ctx.registerComment(s"""Codegend pipeline for\n${child.treeString.trim}""")}` from `child.treeString` to this.treeString`, which will include the codegen stage ID through the `treeString`, just like the explain output.
2. Simply add `$codegenStageId` into the `Codegend pipeline for` line.
3. Do both above. (Did I say two...?)

Which do you prefer?

Reply:
Thank for your changes. They look very good.

I missed one fact (I was sleepy :)). `ctx.registerComment()` is enabled only when `spark.sql.codegen.comments` is `true`. It would be good to add the id in the comment regardless of `spark.sql.codegen.comments` since this comment is very small.
I could create a follow-up PR this afternoon.

Reply:
Yes, I like that idea too. Please ping me on the follow-up PR as well. Thanks!